daml/ContingentClaims/Observable.daml

```diff
@@ -3,45 +3,27 @@
 -- SPDX-License-Identifier: Apache-2.0
 --
 
-{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE UndecidableInstances #-}
 
 module ContingentClaims.Observable where
 
-import Prelude (Bool, Decimal)
-
-infixl 6 +
-infixl 6 -
-infixl 7 *
-infixl 7 /
-
-type Observable f t = (TimeF f t,
-  PointF f t t,
-  PointF f t Decimal,
-  RingF f t Decimal,
-  InequalityF f t Decimal,
-  InequalityF f t t)
-
-class TimeF f t where
-  time : f t t
-
-class PointF f t a where
-  pure : a -> f t a
-
-class RingF f t a where
-  zero : f t a
-  one : f t a
-  (+) : f t a -> f t a -> f t a
-  negate : f t a -> f t a
-  (-) : f t a -> f t a -> f t a
-  x - y = x + ContingentClaims.Observable.negate y
-  (*) : f t a -> f t a -> f t a
-  (/) : f t a -> f t a -> f t a
-
-class InequalityF f t a where
-  (<=) : f t a -> f t a -> f t Bool
-  (==) : f t a -> f t a -> f t Bool
-
-{-
-class LogicF f where
-  (&&) : f Bool -> f Bool -> f Bool
--}
+import Prelude hiding (pure)
+import Daml.Control.Arrow (Kleisli(..))
+
+type Key = Text
+
+type Observable f t x = (Point f t x , Number (f t x), Divisible (f t x))
+
+data Inequality f t x = TimeLte t | Lte (f t x, f t x)
+
+deriving instance (Eq t, Eq (f t x)) => Eq (Inequality f t x)
+
+deriving instance (Show t, Show (f t x)) => Show (Inequality f t x)
+
+class Point f t x where
+  pure : x -> f t x
+  observe : Key -> f t x
+
+class Interpret f where
```

Comment:
This needs a better name. When we add other interpreters, what will we call them?

`eval` is unsafe - it will fail when we try to lookup an Observable that is not yet known.
Compare this with valuation of the claim, which could also be considered `eval`uation, but which doesn't fail.